cmd/podman/libpodruntime/runtime.go

```diff
@@ -1,17 +1,54 @@
 package libpodruntime
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
 	"github.com/containers/storage"
 	"github.com/projectatomic/libpod/libpod"
 	"github.com/urfave/cli"
 )
 
 // GetRuntime generates a new libpod runtime configured by command line options
 func GetRuntime(c *cli.Context) (*libpod.Runtime, error) {
-	storageOpts := storage.DefaultStoreOptions
+	storageOpts, err := GetDefaultStoreOptions()
+	if err != nil {
+		return nil, err
+	}
 	return GetRuntimeWithStorageOpts(c, &storageOpts)
 }
 
+func GetRootlessStorageOpts() (storage.StoreOptions, error) {
+	var opts storage.StoreOptions
+
+	runtimePath := os.Getenv("XDG_RUNTIME_DIR")
+	if runtimePath == "" {
+		return opts, fmt.Errorf("XDG_RUNTIME_DIR not specified")
+	}
+	home := os.Getenv("HOME")
+	if home == "" {
+		return opts, fmt.Errorf("HOME not specified")
+	}
+
+	opts.RunRoot = filepath.Join(runtimePath, "run")
+	opts.GraphRoot = filepath.Join(home, ".containers", "storage")
```

Comment:
`XDG_DATA_DIR`? (`~/.local/share/containers`)

Reply:
yes, good idea.  I've changed the patch to honor `XDG_DATA_DIR` when specified.

I've also opened a PR for atomic so we do the same thing with the atomic containers:

https://github.com/projectatomic/atomic/pull/1240